command/command.go

```diff
@@ -0,0 +1,85 @@
+package command
+
+import (
+	"fmt"
+	"log"
+	"oionetdata/netdata"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+type Command struct {
+	Desc   string
+	Family string
+	Cmd    string
+}
+
+type Worker interface {
+	SinceLastRun() time.Duration
+	AddChart(chart *netdata.Chart)
+}
+
+type collector struct {
+	cmds        map[string]Command
+	data        map[string]string
+	cache       map[string]bool
+	wait        time.Duration
+	cmdInterval time.Duration
+	worker      Worker
+}
+
+func NewCollector(cmds map[string]Command, cmdInterval int64, w Worker) *collector {
+	return &collector{
+		cmds:        cmds,
+		cmdInterval: time.Duration(cmdInterval) * time.Second,
+		data:        nil,
+		wait:        time.Duration(0 * time.Second),
+		cache:       make(map[string]bool),
+		worker:      w,
+	}
+}
+
+func (c *collector) Collect() (map[string]string, error) {
+
+	c.wait -= c.worker.SinceLastRun()
```

Comment:
I don't understant why the collector needs to access this value from the worker.